dev/com.ibm.ws.jca_fat/test-applications/fvtweb/src/web/JCAFVTServlet.java

```diff
@@ -467,6 +470,57 @@ public void testMaxPoolSize2(HttpServletRequest request, HttpServletResponse res
         }
     }
 
+    /**
+     * Make sure invalid connections are getting cleaned up properly when they go through the non-optimal free pool check
+     */
+    public void testNonOptimalFreePoolInvalidConnectionCleanup(HttpServletRequest request, HttpServletResponse response) throws Throwable {
+        FVTConnectionFactory cf = (FVTConnectionFactory) new InitialContext().lookup("java:comp/env/jms/cf1");
+        UserTransaction tran = (UserTransaction) new InitialContext().lookup("java:comp/UserTransaction");
+        FVTConnection con1 = null;
+        FVTConnection con2 = null;
+        FVTManagedConnection mc1 = null;
+        FVTManagedConnection mc2 = null;
+
+        try {
+            tran.begin();
+            con1 = (FVTConnection) cf.createConnection("user1", "pwd1");
+            con2 = (FVTConnection) cf.createConnection("user2", "pwd2");
+        } catch (JMSException x) {
```

Comment:
These catches are a bit dangerous because they suppress the exception.  

If catching an exception should be a failure state then you should have the method throw the error.

If you expect an exception to be thrown and caught then you could should fail if no exception was thrown.

Reply:
Good point.  Those catch blocks should not be reached, but should throw an exception to help decipher why it ended up there.  Thanks.